Automattic/wp-calypso
# Add: App Banner Visibility state - hide banner in sidebar and post error views

Add App Banner Visibility to ui state. 

This allows us to have more fine grane control of when the App Banner should be visible without having to pass that info into the AppBanner component. 

#### Testing instructions

1. Visit the reader on a logged in mobile device. Or use the browser dev tools to visit the same page in the in the responsive mode. You should see the Apps Banner.
2. Open the sidebar in the Reader by clicking the "Read" icon in the masterbar.
- Notice that the Apps Banner doesn't show up.
3. Visit a single post in the reader. Notice that the App  Banner shows up. 
- Visit a post that doesn't exist. Notice that the App Banner is not visible any more once the page loads. 


Before:
<img width="360" alt="Screen Shot 2021-10-04 at 5 14 30 PM" src="https://user-images.githubusercontent.com/115071/135941107-7eea2194-9bdb-42c0-9b55-e890f1886d5c.png">
<img width="360" alt="Screen Shot 2021-10-04 at 5 14 13 PM" src="https://user-images.githubusercontent.com/115071/135941108-d49fbaaf-af20-484b-98bd-e6a4efad6342.png">

After:
<img width="360" alt="Screen Shot 2021-10-04 at 5 12 54 PM" src="https://user-images.githubusercontent.com/115071/135941127-33883256-eb30-4b8e-95b1-8bfac5480bde.png">
<img width="360" alt="Screen Shot 2021-10-04 at 5 13 25 PM" src="https://user-images.githubusercontent.com/115071/135941125-7882fd28-7a94-4adc-be67-8922ae74251a.png">
